Move functions in locallib.php to a class object groups

kei178 opened the pull request:
## Description of the change

Moved the following functions in locallib.php to a new class object `groups.php`:

- `accredible_get_groups` -> `groups->get_groups`
- `accredible_get_templates` -> `groups->get_templates`
- `sync_course_with_accredible` -> `groups->sync_group_with`

## Coding style

The new class `groups.php` meets the Moodle coding style except for `Variable "apiRest" must be all lower-case` which will be addressed when refactoring `apiRest.php`.

## How to test

The new functions are covered by unit tests. You can also manually test the functions by the following steps.

### `groups->get_groups`

1. Go to the Accredible activity edit page.
2. Check if select options of "Cohort name (from dashboard)" are properly displayed.

### `groups->get_templates`

1. Temporarily update the `if` condition at Line116 in `mod_form.php` to return TRUE.
2. Go to the Accredible activity edit page.
3. Check if select options of "Group" are properly displayed.

### `groups->sync_group_with`

#### When `instanceid` is NOT null (update_group)

1. Go to the Accredible activity edit page.
2. Change a different "Cohort name (from dashboard)" and save.
3. Check if the Accredible activity is linked to the selected _Group_ in Accredible.

#### When `instanceid` is null (create_group)

1. Create a new Accredible activity.
2. Check if the new Accredible activity is successfully created as well as the _Group_ in Accredible.

naveenbalachandar reviewed (changes requested):
<img width="827" alt="Screenshot 2021-09-28 at 7 31 18 PM" src="https://user-images.githubusercontent.com/77011560/135102229-8081c6e9-7710-4df1-9879-8d15b2dd0803.png">
@kei178 The code-checker gives out some errors for groups_test.php
Could you please take a look at this as well.

kei178 commented on classes/local/groups.php:
```diff
@@ -0,0 +1,138 @@
+     * @return array[stdClass] $templates
+     */
+    public function get_templates() {
+        $response = $this->apiRest->search_groups(10000, 1);
```
'search_groups' has already been covered by existing unit tests.

kei178 commented:
@naveenbalachandar - I added unit tests for the group-related functions in apiRest.php. I also updated the coding style in groups_test.php.

<img width="719" alt="Screen Shot 2021-09-29 at 10 08 44" src="https://user-images.githubusercontent.com/20009662/135228855-9d871eb3-4820-40d8-9a6f-c2f7d83c247c.png">

kei178 commented:
@naveenbalachandar - Fixed the coding style error in apiRest_test.php.

naveenbalachandar reviewed (approved):
LGTM!
